lib/model-access-wrapper.js

```diff
@@ -604,10 +622,23 @@ class ModelAccessWrapper extends CrispHooks {
 	 * @param {UnimodelDocument} doc - Initial document to create diff from
 	 * @param {Object} data - Changed data to diff against `doc`
 	 */
-	_checkDiffPermissions(permissions, doc, data) {
+	_checkDiffPermissions(permissions, doc, data, overwriteProtected = false) {
 		const grant = permissions.getTargetGrant(this.options.permissionsTarget, data);
 		const patch = Update.createFromDiff(doc.data, data);
-		const fields = _.pull(createUpdate(patch).getUpdatedFields(), '_id');
+		const schema = this.model.getSchema();
+		let fields = createUpdate(patch).getUpdatedFields();
+
+		// Skip permissions checks on protected fields, as they should not exist
+		if (overwriteProtected) {
+			grant.check('overwriteProtected');
+		} else if (this.hasProtectedFields) {
+			// Make sure protected fields have been stripped out from the target doc by this point
+			if (this.protectedFields.some((field) => getPath(data, field) !== undefined)) {
```

Comment:
This won't behave as expected with arrays.  The `this.protectedFields` array comes from a `traverseSchema` call above, and paths in common-schema are fully-qualified (ie, they look like "array.$.field").  Passing a path like this to `getPath()` will try to fetch the literal field called "array.$.field".  There may be a few other cases in this file where arrays will behave incorrectly for this reason.